apis/kueue/v1alpha2/clusterqueue_types.go

```diff
@@ -243,10 +243,63 @@ type ClusterQueueStatus struct {
 	// clusterQueue and haven't finished yet.
 	// +optional
 	AdmittedWorkloads int32 `json:"admittedWorkloads"`
+
+	// conditions hold the latest available observations of the ClusterQueue
+	// current state.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []ClusterQueueCondition `json:"conditions,omitempty"`
```

Comment:
```suggestion
	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

We are using that struct everywhere else already.

Reply:
We need to use `ClusterQueueConditions` instead of `metav1.Condition` to hold the `ClusterQueueConditionType`.
Or, is it better to hold `ClusterQueueConditionType` in `ClusterQueueStatus` like the following?

```go
type ClusterQueueStatus struct {
	// usedResources are the resources (by flavor) currently in use by the
	// workloads assigned to this clusterQueue.
	// +optional
	UsedResources UsedResources `json:"usedResources"`

	// PendingWorkloads is the number of workloads currently waiting to be
	// admitted to this clusterQueue.
	// +optional
	PendingWorkloads int32 `json:"pendingWorkloads"`

	// AdmittedWorkloads is the number of workloads currently admitted to this
	// clusterQueue and haven't finished yet.
	// +optional
	AdmittedWorkloads int32 `json:"admittedWorkloads"`

	// type of condition could be:
	//
	// ClusterQueuePending: clusterQueue is accepted but not yet active.
	// In this state, it can't admit new workloads and its quota can't be borrowed
	// by other active ClusterQueues in the same cohort.
	//
	// ClusterQueueActive: clusterQueue can admit new workloads and its quota
	// can be borrowed by other ClusterQueues in the same cohort.
	Type ClusterQueueConditionType `json:"type"`

	// conditions hold the latest available observations of the ClusterQueue
	// current state.
	// +optional
	// +listType=map
	// +listMapKey=type
	Conditions []metav1.Condition `json:"conditions,omitempty"`
}
```

Reply:
Yeah, it's not worth the extra maintenance effort just to have the type. strings are fine.

Reply:
It makes sense. I will change to the following API. And I add `Active` or `Pending` into metav1.Condition.Type.

```go
type ClusterQueueStatus struct {
	// usedResources are the resources (by flavor) currently in use by the
	// workloads assigned to this clusterQueue.
	// +optional
	UsedResources UsedResources `json:"usedResources"`

	// PendingWorkloads is the number of workloads currently waiting to be
	// admitted to this clusterQueue.
	// +optional
	PendingWorkloads int32 `json:"pendingWorkloads"`

	// AdmittedWorkloads is the number of workloads currently admitted to this
	// clusterQueue and haven't finished yet.
	// +optional
	AdmittedWorkloads int32 `json:"admittedWorkloads"`

	// conditions hold the latest available observations of the ClusterQueue
	// current state.
	// +optional
	// +listType=map
	// +listMapKey=type
	Conditions []metav1.Condition `json:"conditions,omitempty"`
}
```

Reply:
that should work, thanks